scipy/stats/_fit.py

```diff
@@ -1317,6 +1355,13 @@ def _gof_iv(dist, data, known_params, fit_params, guessed_params, statistic,
         message = "`n_mc_samples` must be an integer."
         raise TypeError(message)
 
+    if statistic == 'sf':
+        message = ("All parameters of the null hypothesized distribution must "
+                   "be specified in `known_params` to use the Shapiro-Francia "
+                   "statistic.")
```

Comment:
why not put the string literal direct;y into the Error/exception?

Reply:
Personal preference for dealing with the line-length limitations of PEP8. The `raise NotImplementedError` and pre-indentation has already eaten up half of our characters limit.

Personally, I also don't mind refining the message just _before_ the branching statement because, IMO, it prepares the reader for what the branching statement is going to check.